libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -107,7 +107,7 @@ uint32_t AP_ESC_Telem::get_active_esc_mask() const {
             // have never seen telem from this ESC
             continue;
         }
-        if (now - _telem_data[i].last_update_ms >= ESC_TELEM_DATA_TIMEOUT_MS
+        if (_telem_data[i].stale(now)
```

Comment:
please can we either make this match the style of rpm_data_within_timeout() or convert that function to this style. The current approach is for all the functionality to live in the front-end so my preference would be to preserve that.

Reply:
I deliberately didn't touch the rpm_data_within_timeout() as I don't yet know how to fix it. I'm pretty sure it has two race condition in it now:
 -  every 71 minutes it will declare a dead RPM source as alive when usec wraps
 - as we have no mutex the last_update_us can update asynchronously, so it could update after we've grabbed AP_HAL::micros(), which means it could declare a source as stale when it isn't, which will cause a notch glitch
I looked at this haven't yet decided on the right way to fix it.
I fixed the millisecond timeout with last_update_ms as that was a bug that was trivial to reproduce (it happens on every boot in sim_vehicle.py) and had a much clearer fix.

Reply:
that code has been merged as part of a different PR which was already approved before this one. Happy to revisit it in a separate PR